src/Integration.Vsix.UnitTests/SonarLintTagger/SonarLanguageRecognizerTests.cs

```diff
@@ -0,0 +1,188 @@
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Linq;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Utilities;
+using Moq;
+using SonarLint.VisualStudio.Integration.Vsix;
+
+namespace SonarLint.VisualStudio.Integration.UnitTests
+{
+    [TestClass]
+    public class SonarLanguageRecognizerTests
+    {
+        private Mock<IContentTypeRegistryService> contentTypeServiceMock;
+        private Mock<IFileExtensionRegistryService> fileExtensionServiceMock;
+        private SonarLanguageRecognizer testSubject;
+        private Mock<ITextDocument> textDocumentMock;
+        private Mock<ITextBuffer> textBufferMock;
+
+       [TestInitialize]
+        public void TestInitialize()
+        {
+            contentTypeServiceMock = new Mock<IContentTypeRegistryService>();
+            fileExtensionServiceMock = new Mock<IFileExtensionRegistryService>();
+            testSubject = new SonarLanguageRecognizer(contentTypeServiceMock.Object, fileExtensionServiceMock.Object);
+
+            textDocumentMock = new Mock<ITextDocument>();
+            textBufferMock = new Mock<ITextBuffer>();
+        }
+
+        [TestMethod]
+        public void Ctor_WhenContentTypeRegistryServiceIsNull_Throws()
+        {
+            // Arrange
+            Action act = () => new SonarLanguageRecognizer(null, new Mock<IFileExtensionRegistryService>().Object);
+
+            // Act & Assert
+            act.ShouldThrow<ArgumentNullException>().And.ParamName.Should().Be("contentTypeRegistryService");
+        }
+
+        [TestMethod]
+        public void Ctor_WhenFileExtensionRegistryServiceIsNull_Throws()
+        {
+            // Arrange
+            Action act = () => new SonarLanguageRecognizer(new Mock<IContentTypeRegistryService>().Object, null);
+
+            // Act & Assert
+            act.ShouldThrow<ArgumentNullException>().And.ParamName.Should().Be("fileExtensionRegistryService");
+        }
+
+        [TestMethod]
+        public void Detect_WhenFileExtensionIsJavascriptRelated_ReturnsJavascript()
+        {
+            foreach (var expectedExtension in new[] { "js", "jsx", "vue" })
```

Comment:
Should the comparison be case-insensitive?

Reply:
Done